homework/module_1-hw_standardlib/task_4.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+import sys
+import argparse
+import os
+import fnmatch
+
+
+def _match_print(collection: list[str], root: str, name: str | None = None):
+    for item in collection:
+        if name is None or fnmatch.fnmatch(item, name):
+            print(os.path.join(root, item))
+
+
+def find(folder, name=None, show_dirs=True, show_files=True):
+    """
+    :param folder: path to a system folder from where to start searching
+    :param name: file/directory name pattern, allows using '*' and '?' symbols
+    :param show_dirs: if True - include directories into search results
+    :param show_files: if True - include files into search results
+    """
```

Comment:
I would add a check that the directory exists

Reply:
Not fixed - this piece of code was given with homework.